### PR TITLE
Enforce that on semgrep#develop always tracks on (or behind) semgrep-interfaces#main

### PR DESCRIPTION
**Pull request by ihji**
test plan: manually tested

### CODE REVIEW

File: .github/workflows/check-submodule.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: check-submodule
```
Comment (aryx) [inline .github/workflows/check-submodule.yml:1]: Maybe add a  comment explaining this is mostly a copy-paste of semgrep-proprietary/.../check-submodule.yml

File: .github/workflows/check-submodule.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+on: [pull_request]
+
+jobs:
+  check-submodule:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout this repo
+        uses: actions/checkout@v3
+      # We need to explicitly check out the Semgrep interfaces submodule since
```
Comment (aryx) [inline .github/workflows/check-submodule.yml]: I think you can just use with: submodules: true
so you don't need to do a separate checkout for semgrep-interfaces, and you can remove this comment about
ssh I don't really understand.

**Review by aryx — APPROVED** [review]

File: .github/workflows/check-submodule.yml (hunk 1 above)
Comment (ihji) [reply .github/workflows/check-submodule.yml:1]: done.

**Comment by aryx** [comment]
how did you manually test this PR?

File: .github/workflows/check-submodule.yml (hunk 2 above)
Comment (ihji) [reply .github/workflows/check-submodule.yml]: Thanks for catching it. semgrep-pro is a private repo, but semgrep-interfaces is public, so a separate checkout should not be necessary.

**Comment by aryx** [comment]
you can add a on: workflow_dispatch
so this workflow can be manually triggered from the Actions tab in github

**Comment by ihji** [comment]
> how did you manually test this PR?

I tested this by manually including the interface submodule update for the non-main branch commit and the main branch commit in this PR. I checked if `check-submodule` failed and succeeded, respectively.

**Comment by aryx** [comment]
How did you trigger the workflow? by making a PR?

**Comment by ihji** [comment]
> How did you trigger the workflow? by making a PR?

by updating the PR.

positive test: https://github.com/semgrep/semgrep/actions/runs/7645999673/job/20833853718
negative test: https://github.com/semgrep/semgrep/actions/runs/7646030086/job/20833947963

**Pull request merged**